File: apps/mobile/lib/domain/entities/budget.dart
enum OwnerType {
  user,
  group,
}

enum BudgetStatus {
  active,
  closed,
  draft,
}

class Budget {
  final int id;
  final OwnerType ownerType;
  final int ownerId;
  final String period; // YYYY-MM
  final int totalAmount;
  final BudgetStatus status;
  final DateTime createdAt;
  final DateTime updatedAt;

  Budget({
    required this.id,
    required this.ownerType,
    required this.ownerId,
    required this.period,
    required this.totalAmount,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  Budget copyWith({
    int? id,
    OwnerType? ownerType,
    int? ownerId,
    String? period,
    int? totalAmount,
    BudgetStatus? status,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Budget(
      id: id ?? this.id,
      ownerType: ownerType ?? this.ownerType,
      ownerId: ownerId ?? this.ownerId,
      period: period ?? this.period,
      totalAmount: totalAmount ?? this.totalAmount,
      status: status ?? this.status,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: household-ledger-backend/src/domain/models/budget.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Copy, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "enum", rename_all = "UPPERCASE")]
pub enum OwnerType {
    User,
    Group,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "enum", rename_all = "UPPERCASE")]
pub enum BudgetStatus {
    Active,
    Closed,
    Draft,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Budget {
    pub id: i64,
    #[sqlx(rename = "owner_type")]
    pub owner_type: OwnerType,
    pub owner_id: i64,
    pub period: String,
    pub total_amount: i64,
    #[sqlx(rename = "status")]
    pub budget_status: BudgetStatus,
    #[sqlx(default)]
    pub created_at: NaiveDateTime,
    #[sqlx(default)]
    pub updated_at: NaiveDateTime,
}

